cmd/time.go

```diff
@@ -0,0 +1,67 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import "time"
+
+var minioTZ = map[string]int{
+
+	"PST": -8 * 3600,
+	"PDT": -7 * 3600,
+
+	"EST": -5 * 3600,
+	"EDT": -4 * 3600,
+
+	"GMT": 0,
+	"UTC": 0,
+
+	"CET":  1 * 3600,
+	"CEST": 2 * 3600,
+}
+
+// parseTime returns the correct time when a timezone is provided
+// with the time string, if the specified timezone is not found in
+// minioTZ, call only the standard golang time parsing.
+func parseTime(layout, value string) (time.Time, error) {
+	// Standard golang time parsing
+	t, err := time.Parse(layout, value)
```

Comment:
> Why can we use https://golang.org/pkg/time/#ParseInLocation?

We can't, ParseInLocation needs location as parameter and the purpose of this PR is to guess the location from the date string. 

For example, if we have this date `Fri, 03 Apr 2015 17:10:00 PST`, golang can parse this string but it doesn't consider PST as it doesn't know it. The parsed date will automatically have UTC timezone which is wrong.